### PR TITLE
Connected to API/ DB

### PR DESCRIPTION
**Pull request by AbdoMallah**
We can now, get all chargers, get a specific charger by id, and update charger status.

### CODE REVIEW

File: lib/ui/bottom_sheets/map_bottom_sheet/snappingcheet.dart [hunk 1]
```diff
@@ -121,36 +123,38 @@ class CustomSnappingSheet extends StatelessWidget {
...
+                        } else {
+                          model.showWideButton = true;
+                        }
+                        return "Invalid charger ID";
+                      } else {
+                        model.chargerCode = input;
+                        model.getChargerById(int.parse(input));
+                        model.showWideButton = true;
+
+                        return '';
```
Comment (diddenordstrom) [inline lib/ui/bottom_sheets/map_bottom_sheet/snappingcheet.dart:138]: A bit hard to read when there are this many conditions.

File: lib/ui/bottom_sheets/map_bottom_sheet/snappingcheet_viewmodel.dart [hunk 2]
```diff
@@ -1,57 +1,94 @@
...
-  set selectedChargerId(int newId) {
-    _selectedChargerId = newId;
+  Color get wideButtonColor {
+    if (selectedCharger.status == 1)
+      return Color.fromRGBO(120, 189, 118, 1);
+    else if (selectedCharger.status == 0)
+      return Color.fromRGBO(239, 96, 72, 1);
+    else
+      return Color.fromRGBO(229, 229, 229, 1);
+  }
```
Comment (diddenordstrom) [inline lib/ui/bottom_sheets/map_bottom_sheet/snappingcheet_viewmodel.dart:40]: I think it is better to write the statements on one line
`if (selectedCharger.status == 1) return Color.fromRGBO();`
`else if (selectedCharger.status == 1) return Color.fromRGBO();`
`else (selectedCharger.status == 1) return Color.fromRGBO();`

File: lib/ui/bottom_sheets/map_bottom_sheet/snappingcheet_viewmodel.dart [hunk 3]
```diff
@@ -1,57 +1,94 @@
...
 
+  String get wideButtonText {
+    if (selectedCharger.status == 1)
+      return 'Begin Charging';
+    else if (selectedCharger.status == 0)
+      return 'Charger Occupied';
+    else
+      return 'Charger Not Identified';
+  }
```
Comment (diddenordstrom) [inline lib/ui/bottom_sheets/map_bottom_sheet/snappingcheet_viewmodel.dart:49]: same as above

Comment (AbdoMallah) [reply lib/ui/bottom_sheets/map_bottom_sheet/snappingcheet_viewmodel.dart:49]: Tack För dina kommentarer Didrik. när det gäller "if statments" så är det (auto indentation )som gör att de tar två rader.

Comment (diddenordstrom) [reply lib/ui/bottom_sheets/map_bottom_sheet/snappingcheet_viewmodel.dart:49]: Misstänkte nästan det :) Bra jobbat!

File: lib/ui/bottom_sheets/map_bottom_sheet/snappingcheet.dart (hunk 1 above)
Comment (AbdoMallah) [reply lib/ui/bottom_sheets/map_bottom_sheet/snappingcheet.dart:138]: Essa Worked on that, I tried my best to make it more readable but I don't want to change another developer code.

**Pull request merged**